common/api/core-bentley.api.md

```diff
@@ -1178,6 +1178,13 @@ export namespace JsonUtils {
 // @public
 export type Listener = (...arg: any[]) => void;
 
+// @beta
```

Comment:
Why beta? What might go wrong? Under what criteria will the type become suitable for promoting to public?

Reply:
I've marked it as `@beta` with the idea to leave some room in case some unexpected use case comes up. I.e. initially I had a naive implementation, where `BeEvent` was used directly, but it wouldn't work with narrowed event types.
I've bumped the type to `@public` since it is unlikely we'll need to change the structure of the type.